File: app/src/main/java/antoninovitale/dropcodechallenge/details/section/viewholder/BaseSectionViewHolder.java
package antoninovitale.dropcodechallenge.details.section.viewholder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import antoninovitale.dropcodechallenge.R;
import butterknife.BindString;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by antoninovitale on 03/09/2017.
 */
class BaseSectionViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.name)
    public TextView name;

    @BindView(R.id.status)
    public TextView status;

    @BindString(R.string.idle)
    public String idle;

    @BindString(R.string.running)
    public String running;

    @BindString(R.string.paused)
    public String paused;

    @BindString(R.string.done)
    public String done;

    BaseSectionViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

}

File: app/src/main/java/antoninovitale/punkapi/app/details/section/viewholder/BaseSectionViewHolder.kt
package antoninovitale.punkapi.app.details.section.viewholder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import antoninovitale.punkapi.app.R

/**
 * Created by antoninovitale on 03/09/2017.
 */
open class BaseSectionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val idle = itemView.context.getString(R.string.idle)
    val running = itemView.context.getString(R.string.running)
    val paused = itemView.context.getString(R.string.paused)
    val done = itemView.context.getString(R.string.done)
}